Clarify that command-line switches need to be manually set in `addprocs`

evetion opened the pull request:
Fixes #50673

also fixed a small typo in an unrelated doc page.

evetion committed:
Document --heap-size-hint and be more clear about addprocs not passing any command line args.

evetion committed:
Fix some typos and link possible exeflags.

nsajko commented:
There's three (!) unrelated changes in this PR. Why don't you separate the other two fixes into separate PRs?

nsajko commented:
Regarding the *main* change, it's a duplicate of #50480

evetion commented:
> There's three (!) unrelated changes in this PR. Why don't you separate the other two fixes into separate PRs?

I'd argue that there is a single main change, documenting how --heap-size-hint (and other flags) work with addproces. That requires adding --heap-size-hint to the docs itself.

The only unrelated change is a one character typo fix that I came across, and I didn't feel it would block this PR, so I included it. I'm happy to split it out to another PR if requested.

> Regarding the _main_ change, it's a duplicate of #50480

I missed this PR by @KnutAM, nice work! Note it's not the _main_ change in this PR, I'd say that's the interaction with addprocs.

fingolfin commented on doc/src/manual/command-line-interface.md:
```diff
@@ -131,6 +131,7 @@ The following is a complete list of command-line switches available when launchi
 |`--track-allocation[={none*\|user\|all}]` |Count bytes allocated by each source line (omitting setting is equivalent to "user")|
 |`--track-allocation=@<path>`           |Count bytes but only in files that fall under the given file path/directory. The `@` prefix is required to select this option. A `@` with no path will track the current directory.|
 |`--bug-report=KIND`                    |Launch a bug report session. It can be used to start a REPL, run a script, or evaluate expressions. It first tries to use BugReporting.jl installed in current environment and falls back to the latest compatible BugReporting.jl if not. For more information, see `--bug-report=help`.|
+|`--heap-size-hint=<size>`              |Forces garbage collection if memory usage is higher than that value. The memory hint might be specified in megabytes (500M) or gigabytes (1G).|
```
In the meantime `--heap-size-hint` was documented at the end of this list, so this here ought to go.
```suggestion
```

fingolfin commented:
Unfortunately this now has a conflict. @evetion would you be willing to resolve that. I think afterwards (and with my other comment taken into account) I think this is in a good shape for merging?

evetion committed:
Update doc/src/manual/command-line-interface.md

Co-authored-by: Max Horn <max@quendi.de>

evetion committed:
Delete stdlib/Distributed/src/managers.jl

evetion commented:
> Unfortunately this now has a conflict. @evetion would you be willing to resolve that. I think afterwards (and with my other comment taken into account) I think this is in a good shape for merging?

Fixed now, and moved part of the PR to the Distributed package.

inkydragon reviewed (commented):
It looks like the current changes have nothing to do with `--heap-size-hint`, maybe a title change is needed, then LGTM.

evetion commented:
> It looks like the current changes have nothing to do with `--heap-size-hint`, maybe a title change is needed, then LGTM.

Thanks for the suggestion, I've updated the title.
